src/JSInterop/Microsoft.JSInterop/src/DotNetObjectRef.cs

```diff
@@ -1,66 +1,21 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Threading;
-
 namespace Microsoft.JSInterop
 {
     /// <summary>
-    /// Wraps a JS interop argument, indicating that the value should not be serialized as JSON
-    /// but instead should be passed as a reference.
-    ///
-    /// To avoid leaking memory, the reference must later be disposed by JS code or by .NET code.
+    /// Provides convenience methods to produce a <see cref="DotNetObjectRef{TValue}" />.
     /// </summary>
-    public class DotNetObjectRef : IDisposable
+    public static class DotNetObjectRef
```

Comment:
Let me know what you think about this design. We don't need the type to be generic, but it avoids cast in user code when receiving it from JS.

```C#
public static void SomeJSInteropMethod(DotNetObjectRef foo)
{
   var value = ((Foo)foo.Value).DoSomething();
}
```

versus 
```C#
public static void SomeJSInteropMethod(DotNetObjectRef<Foo> foo)
{
   var value = foo.Value.DoSomething();
}
```

We also don't need this factory method, but it's a convenient way to avoid a bit of typing when creating these reference instances.

```C#
var ref = new DotNetObjectRef<Foo>(foo);
```

versus

```C#
var ref = DotNetObjectRef.Create(foo);
```

Reply:
I agree on all of these points. Immutable collections does a similar thing and it's decent. @SteveSandersonMS @javiercn any thoughts?

Reply:
This makes sense and looks good!